### DISCUSSION TITLE
ctrl+c handling in kitty keyboard protocol

### DISCUSSION THREAD

**Discussion by metamaterialsuit** [kovidgoyal/kitty#7179] [Q&A]
Quoting from the text:

"This makes it very easy to parse key events in an application. In particular, ctrl+c will no longer generate the SIGINT signal, but instead be delivered as a CSI u escape code. This has the nice side effect of making it much easier to integrate into the application event loop."

I was wondering whether this could be explained further.  Do you mean that a user generated control-c will not cause kitty to generate a sigint? Is that not behavior that we want? And if, alternatively, you mean that the application sending a control+c would not generate an interrupt, can you explain why this would happen?

**Comment by kovidgoyal** [comment]
On Fri, Mar 01, 2024 at 07:07:05PM -0800, metamaterialsuit wrote:
> Quoting from the text:
> 
> "This makes it very easy to parse key events in an application. In particular, ctrl+c will no longer generate the SIGINT signal, but instead be delivered as a CSI u escape code. This has the nice side effect of making it much easier to integrate into the application event loop."
> 
> I was wondering whether this could be explained further.  Do you mean that a user generated control-c will not cause kitty to generate a sigint? 

Yes. And no, Ctrl-C generating signals is not behavior we want for any
non-trivial terminal mode program. These will have event loops, signals
dont integrate well with event loops, the only way to integrate them is
to have a signal handler that writes to a pipe which is watched in the
event loop. At which point it is easier to simply handle the key press
yourself.

> Is that not behavior that we want? And if, alternatively, you mean that the application sending a control+c would not generate an interrupt, can you explain why this would happen? 

Applications cannot generate interrupts by sending ctrl+c

**Reply by metamaterialsuit** [reply]
Ah, ok, thank you!
